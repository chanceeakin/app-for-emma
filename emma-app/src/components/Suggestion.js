//@flow
import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import type { Suggestion } from './../types/Suggestions.js.flow'
import { pageLayout, font, text } from './../styles'

type Props = {
  suggestions: Suggestion,
  isFontLoaded: Boolean,
};

const styles = StyleSheet.create({
  row: pageLayout.row,
  title: text.title,
  description: text.description,
  font: font.skia,
})

const SuggestionComp = ({ suggestions, isFontLoaded }: Props) => (
  <View style={styles.row}>
    {suggestions.title ? (
      <Text style={[styles.title, isFontLoaded ? styles.font : null]}>
        {suggestions.title}
      </Text>
    ) : null}
    {suggestions.description ? (
      <Text style={[styles.description, styles.font]}>
        {suggestions.description}
      </Text>
    ) : null}
  </View>
)

export default SuggestionComp
