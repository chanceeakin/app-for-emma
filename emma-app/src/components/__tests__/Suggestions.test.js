import React from 'react'
import Suggestion from '../Suggestion'

import renderer from 'react-test-renderer'

const props = {
  suggestions: {
    title: 'hi',
    description: 'person actually reading my tests',
  },
}

describe('Suggestions Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Suggestion {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
