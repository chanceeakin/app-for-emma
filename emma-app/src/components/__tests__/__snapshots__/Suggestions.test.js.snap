// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions Component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "minHeight": 100,
      "padding": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#0D1B2A",
          "fontSize": 72,
          "textAlign": "center",
        },
        null,
      ]
    }
  >
    hi
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4C3B4D",
          "fontSize": 30,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "Skia",
        },
      ]
    }
  >
    person actually reading my tests
  </Text>
</View>
`;
